sdk/go/client.go

```diff
@@ -49,6 +50,31 @@ func (fc *GrpcClient) GetOnlineFeatures(ctx context.Context, req *OnlineFeatures
 		return nil, err
 	}
 	resp, err := fc.cli.GetOnlineFeatures(ctx, featuresRequest)
+	
+	// collect unqiue entity refs from entity rows
+	var entityRefs map[string]struct{}
+	for _, entityRows := range req.Entities {
+		for ref, _ := range entityRows {
+			entityRefs[ref] = struct{}{}
+		}
+	}
+
+	// strip projects from to projects
```

Comment:
Why does the SDK strip projects from serving's response?

Reply:
We want to remove projects in Feature References in user facing SDK API's completely.
- If the user makes a request and specifies the `project` parameter on get online/batch features, the SDK will add the project to the Feature Reference submitted to serving.
- Serving processes these Feature References with projects and keys the fields with Feature References with Projects unaware that the project was added by the SDK not the user.
- The SDK strips the Feature References of their projects and returns them to the user.

The stripping of projects is done so that when the user specifies `driver_fs:name` with `project="rides
` they would get `driver_fs:name` back instead of `rides/driver_fs:name` returned from serving.

Reply:
I think rather than strip projects completely it would be a lot more intuitive to the user if the returned feature reference were consistent with what they provided (if a user provides `rides/driver_fs:name` they would expect to retrieve their values with the reference `rides/driver_fs:name` over `driver_fs:name`)

Reply:
Yes that would be indeed clearer if that is the case.

However this PR attempts to address #631 which stipulates that we want to move towards to project less Feature References as we figure what place projects has in Feast core model in #479. 

As such in the new `v0.5` SDKs, I have dropped projects in Feature References entirely,  users can only specify `driver_fs:name` or `name`. Putting in feature references (ie `rides/driver_fs:name` would result in an error in the SDK. If users want to specify a project, they have to do so via a dedicated parameter in get online/batch features methods. 

This is done in the SDK instead of Serving as I also want to maintain some form of compatibility with the old v0.4 SDK which does have project in its feature references.

Reply:
So that means in this transitional state retrieving across projects is not possible?

Reply:
Yes